packages/reporters/cli/src/SimpleCLIReporter.js

```diff
@@ -0,0 +1,103 @@
+// @flow strict-local
+
+import type {ReporterEvent, ParcelOptions} from '@parcel/types';
+
+import type {Writable} from 'stream';
+
+import {render} from 'ink';
+import {Reporter} from '@parcel/plugin';
+import * as React from 'react';
+import prettifyTime from '@parcel/utils/src/prettifyTime';
+
+import BundleReport from './BundleReport';
+import prettyError from './prettyError';
+import {getProgressMessage} from './utils';
+import logLevels from './logLevels';
+
+export default new Reporter({
+  report(event: ReporterEvent, options: ParcelOptions) {
+    _report(event, options);
+  }
+});
+
+let stdout = process.stdout;
+let stderr = process.stderr;
+
+// Exported only for test
+export function _setStdio(stdoutLike: Writable, stderrLike: Writable) {
+  stdout = stdoutLike;
+  stderr = stderrLike;
+}
+
+// Exported only for test
+export function _report(event: ReporterEvent, options: ParcelOptions): void {
+  let logLevelFilter = logLevels[options.logLevel || 'info'];
+
+  switch (event.type) {
+    case 'buildProgress':
+      if (logLevelFilter < logLevels.info) {
+        break;
+      }
+
+      writeOut(getProgressMessage(event));
+      break;
+    case 'buildSuccess':
+      if (logLevelFilter < logLevels.info) {
+        break;
+      }
+
+      writeOut(`Built in ${prettifyTime(event.buildTime)}`);
+      if (options.mode === 'production') {
+        render(<BundleReport bundleGraph={event.bundleGraph} />);
+      }
+      break;
+    case 'buildFailure':
+      if (logLevelFilter < logLevels.error) {
+        break;
+      }
+
+      writeErr(event.error);
+      break;
+    case 'log': {
+      switch (event.level) {
+        case 'warn':
+        case 'error':
+          if (logLevelFilter >= logLevels[event.level]) {
+            writeErr(event.message);
+          }
+          break;
+        case 'info':
+        case 'verbose':
+        case 'progress':
+        case 'success':
+          if (logLevelFilter >= logLevels[event.level]) {
+            writeOut(event.message);
+          }
+          break;
+        default:
+          throw new Error('Unknown log level ' + event.level);
+      }
+    }
+  }
+}
+
+function writeOut(message: ?string): void {
+  if (message != null) {
+    stdout.write(message + '\n');
```

Comment:
using `console.log`/`console.error` rather than stdio would make this more portable, but we're already kind of coupled to stdio with ink.